In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv(r'C:\Users\Lenovo\Downloads\Job Market Insights\InternshalaJobs.csv')
df

Unnamed: 0                                        Position  \
0             0                        Junior Interior Designer   
1             1  Business Development Representative - B2B SaaS   
2             2                      Corporate Sales Specialist   
3             3                  Career Guidance Expert - Sales   
4             4                           Video Content Creator   
..          ...                                             ...   
404         404                  Business Development Executive   
405         405                           Engagement Strategist   
406         406                    Brand Partnerships Associate   
407         407                  Business Development Executive   
408         408                Junior Lead Generation Associate   

                                        Company Name        Location  \
0                           Dexterio Private Limited          Jaipur   
1            Zylu Business Solutions Private Limited       Bangalore   
2                         Hot Stuffs Private Limited          Mumbai   
3    NxtWave Disruptive Technologies Private Limited  Work from home   
4             ReThought Technologies Private Limited           Delhi   
..                                               ...             ...   
404                Mohite Consultancy Services (MCS)     Koramangala   
405                                        OpeninApp       Bangalore   
406                                       Confluencr       Bangalore   
407                                    Nerdy Academy          Mumbai   
408                   Fabulous Media Private Limited         Gurgaon   

             Start Date           CTC (Annual) Experience   Posted Ago Role  
0    Starts Immediately  ₹ 2,00,000 - 2,40,000  0-2 years  3 weeks ago  Job  
1    Starts Immediately  ₹ 3,00,000 - 5,00,000  0-5 years    1 day ago  Job  
2    Starts Immediately  ₹ 2,10,000 - 3,00,000  0-2 years   6 days ago  Job  
3    Starts Immediately  ₹ 3,00,000 - 6,00,000  0-5 years  2 weeks ago  Job  
4    Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years  2 weeks ago  Job  
..                  ...                    ...        ...          ...  ...  
404  Starts Immediately  ₹ 2,16,000 - 2,40,000  0-2 years   1 week ago  Job  
405  Starts Immediately  ₹ 2,00,000 - 8,00,000  0-5 years   1 week ago  Job  
406  Starts Immediately  ₹ 3,00,000 - 4,50,000  0-2 years   6 days ago  Job  
407  Starts Immediately  ₹ 2,40,000 - 5,80,000  0-5 years  3 weeks ago  Job  
408  Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years   5 days ago  Job  

[409 rows x 9 columns]

# To display all 409 rows :
> pd.set_option('display.max_rows', None)

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 409 entries, 0 to 408
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Unnamed: 0    409 non-null    int64 
 1   Position      409 non-null    object
 2   Company Name  409 non-null    object
 3   Location      409 non-null    object
 4   Start Date    409 non-null    object
 5   CTC (Annual)  409 non-null    object
 6   Experience    409 non-null    object
 7   Posted Ago    409 non-null    object
 8   Role          409 non-null    object
dtypes: int64(1), object(8)
memory usage: 28.9+ KB


In [4]:
df.describe()

Unnamed: 0
count  409.000000
mean   204.000000
std    118.212379
min      0.000000
25%    102.000000
50%    204.000000
75%    306.000000
max    408.000000

In [5]:
df.describe(include='all')

Unnamed: 0                        Position        Company Name  \
count   409.000000                             409                 409   
unique         NaN                             282                 242   
top            NaN  Business Development Executive  Zigsaw Consultancy   
freq           NaN                              27                  10   
mean    204.000000                             NaN                 NaN   
std     118.212379                             NaN                 NaN   
min       0.000000                             NaN                 NaN   
25%     102.000000                             NaN                 NaN   
50%     204.000000                             NaN                 NaN   
75%     306.000000                             NaN                 NaN   
max     408.000000                             NaN                 NaN   

       Location          Start Date           CTC (Annual) Experience  \
count       409                 409                    409        409   
unique       92                   1                    114         15   
top      Mumbai  Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years   
freq         72                 409                     48        289   
mean        NaN                 NaN                    NaN        NaN   
std         NaN                 NaN                    NaN        NaN   
min         NaN                 NaN                    NaN        NaN   
25%         NaN                 NaN                    NaN        NaN   
50%         NaN                 NaN                    NaN        NaN   
75%         NaN                 NaN                    NaN        NaN   
max         NaN                 NaN                    NaN        NaN   

        Posted Ago Role  
count          409  409  
unique          12    2  
top     1 week ago  Job  
freq            58  407  
mean           NaN  NaN  
std            NaN  NaN  
min            NaN  NaN  
25%            NaN  NaN  
50%            NaN  NaN  
75%            NaN  NaN  
max            NaN  NaN

In [6]:
df.describe(include='object')

Position        Company Name Location  \
count                              409                 409      409   
unique                             282                 242       92   
top     Business Development Executive  Zigsaw Consultancy   Mumbai   
freq                                27                  10       72   

                Start Date           CTC (Annual) Experience  Posted Ago Role  
count                  409                    409        409         409  409  
unique                   1                    114         15          12    2  
top     Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years  1 week ago  Job  
freq                   409                     48        289          58  407

In [7]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

# Duplicate Values

In [8]:
df.duplicated()
df.duplicated(keep = 'first')
df.duplicated(keep = 'last')

0      False
1      False
2      False
3      False
4      False
       ...  
404    False
405    False
406    False
407    False
408    False
Length: 409, dtype: bool

0      False
1      False
2      False
3      False
4      False
       ...  
404    False
405    False
406    False
407    False
408    False
Length: 409, dtype: bool

0      False
1      False
2      False
3      False
4      False
       ...  
404    False
405    False
406    False
407    False
408    False
Length: 409, dtype: bool

In [9]:
a = df.drop_duplicates()
b = df.drop_duplicates(keep = 'first')
c = df.drop_duplicates(keep = 'last')
a
b
c

Unnamed: 0                                        Position  \
0             0                        Junior Interior Designer   
1             1  Business Development Representative - B2B SaaS   
2             2                      Corporate Sales Specialist   
3             3                  Career Guidance Expert - Sales   
4             4                           Video Content Creator   
..          ...                                             ...   
404         404                  Business Development Executive   
405         405                           Engagement Strategist   
406         406                    Brand Partnerships Associate   
407         407                  Business Development Executive   
408         408                Junior Lead Generation Associate   

                                        Company Name        Location  \
0                           Dexterio Private Limited          Jaipur   
1            Zylu Business Solutions Private Limited       Bangalore   
2                         Hot Stuffs Private Limited          Mumbai   
3    NxtWave Disruptive Technologies Private Limited  Work from home   
4             ReThought Technologies Private Limited           Delhi   
..                                               ...             ...   
404                Mohite Consultancy Services (MCS)     Koramangala   
405                                        OpeninApp       Bangalore   
406                                       Confluencr       Bangalore   
407                                    Nerdy Academy          Mumbai   
408                   Fabulous Media Private Limited         Gurgaon   

             Start Date           CTC (Annual) Experience   Posted Ago Role  
0    Starts Immediately  ₹ 2,00,000 - 2,40,000  0-2 years  3 weeks ago  Job  
1    Starts Immediately  ₹ 3,00,000 - 5,00,000  0-5 years    1 day ago  Job  
2    Starts Immediately  ₹ 2,10,000 - 3,00,000  0-2 years   6 days ago  Job  
3    Starts Immediately  ₹ 3,00,000 - 6,00,000  0-5 years  2 weeks ago  Job  
4    Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years  2 weeks ago  Job  
..                  ...                    ...        ...          ...  ...  
404  Starts Immediately  ₹ 2,16,000 - 2,40,000  0-2 years   1 week ago  Job  
405  Starts Immediately  ₹ 2,00,000 - 8,00,000  0-5 years   1 week ago  Job  
406  Starts Immediately  ₹ 3,00,000 - 4,50,000  0-2 years   6 days ago  Job  
407  Starts Immediately  ₹ 2,40,000 - 5,80,000  0-5 years  3 weeks ago  Job  
408  Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years   5 days ago  Job  

[409 rows x 9 columns]

Unnamed: 0                                        Position  \
0             0                        Junior Interior Designer   
1             1  Business Development Representative - B2B SaaS   
2             2                      Corporate Sales Specialist   
3             3                  Career Guidance Expert - Sales   
4             4                           Video Content Creator   
..          ...                                             ...   
404         404                  Business Development Executive   
405         405                           Engagement Strategist   
406         406                    Brand Partnerships Associate   
407         407                  Business Development Executive   
408         408                Junior Lead Generation Associate   

                                        Company Name        Location  \
0                           Dexterio Private Limited          Jaipur   
1            Zylu Business Solutions Private Limited       Bangalore   
2                         Hot Stuffs Private Limited          Mumbai   
3    NxtWave Disruptive Technologies Private Limited  Work from home   
4             ReThought Technologies Private Limited           Delhi   
..                                               ...             ...   
404                Mohite Consultancy Services (MCS)     Koramangala   
405                                        OpeninApp       Bangalore   
406                                       Confluencr       Bangalore   
407                                    Nerdy Academy          Mumbai   
408                   Fabulous Media Private Limited         Gurgaon   

             Start Date           CTC (Annual) Experience   Posted Ago Role  
0    Starts Immediately  ₹ 2,00,000 - 2,40,000  0-2 years  3 weeks ago  Job  
1    Starts Immediately  ₹ 3,00,000 - 5,00,000  0-5 years    1 day ago  Job  
2    Starts Immediately  ₹ 2,10,000 - 3,00,000  0-2 years   6 days ago  Job  
3    Starts Immediately  ₹ 3,00,000 - 6,00,000  0-5 years  2 weeks ago  Job  
4    Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years  2 weeks ago  Job  
..                  ...                    ...        ...          ...  ...  
404  Starts Immediately  ₹ 2,16,000 - 2,40,000  0-2 years   1 week ago  Job  
405  Starts Immediately  ₹ 2,00,000 - 8,00,000  0-5 years   1 week ago  Job  
406  Starts Immediately  ₹ 3,00,000 - 4,50,000  0-2 years   6 days ago  Job  
407  Starts Immediately  ₹ 2,40,000 - 5,80,000  0-5 years  3 weeks ago  Job  
408  Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years   5 days ago  Job  

[409 rows x 9 columns]

Unnamed: 0                                        Position  \
0             0                        Junior Interior Designer   
1             1  Business Development Representative - B2B SaaS   
2             2                      Corporate Sales Specialist   
3             3                  Career Guidance Expert - Sales   
4             4                           Video Content Creator   
..          ...                                             ...   
404         404                  Business Development Executive   
405         405                           Engagement Strategist   
406         406                    Brand Partnerships Associate   
407         407                  Business Development Executive   
408         408                Junior Lead Generation Associate   

                                        Company Name        Location  \
0                           Dexterio Private Limited          Jaipur   
1            Zylu Business Solutions Private Limited       Bangalore   
2                         Hot Stuffs Private Limited          Mumbai   
3    NxtWave Disruptive Technologies Private Limited  Work from home   
4             ReThought Technologies Private Limited           Delhi   
..                                               ...             ...   
404                Mohite Consultancy Services (MCS)     Koramangala   
405                                        OpeninApp       Bangalore   
406                                       Confluencr       Bangalore   
407                                    Nerdy Academy          Mumbai   
408                   Fabulous Media Private Limited         Gurgaon   

             Start Date           CTC (Annual) Experience   Posted Ago Role  
0    Starts Immediately  ₹ 2,00,000 - 2,40,000  0-2 years  3 weeks ago  Job  
1    Starts Immediately  ₹ 3,00,000 - 5,00,000  0-5 years    1 day ago  Job  
2    Starts Immediately  ₹ 2,10,000 - 3,00,000  0-2 years   6 days ago  Job  
3    Starts Immediately  ₹ 3,00,000 - 6,00,000  0-5 years  2 weeks ago  Job  
4    Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years  2 weeks ago  Job  
..                  ...                    ...        ...          ...  ...  
404  Starts Immediately  ₹ 2,16,000 - 2,40,000  0-2 years   1 week ago  Job  
405  Starts Immediately  ₹ 2,00,000 - 8,00,000  0-5 years   1 week ago  Job  
406  Starts Immediately  ₹ 3,00,000 - 4,50,000  0-2 years   6 days ago  Job  
407  Starts Immediately  ₹ 2,40,000 - 5,80,000  0-5 years  3 weeks ago  Job  
408  Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years   5 days ago  Job  

[409 rows x 9 columns]

In [10]:
a.reset_index(drop = True)
b.reset_index(drop = True)
c.reset_index(drop = True)

Unnamed: 0                                        Position  \
0             0                        Junior Interior Designer   
1             1  Business Development Representative - B2B SaaS   
2             2                      Corporate Sales Specialist   
3             3                  Career Guidance Expert - Sales   
4             4                           Video Content Creator   
..          ...                                             ...   
404         404                  Business Development Executive   
405         405                           Engagement Strategist   
406         406                    Brand Partnerships Associate   
407         407                  Business Development Executive   
408         408                Junior Lead Generation Associate   

                                        Company Name        Location  \
0                           Dexterio Private Limited          Jaipur   
1            Zylu Business Solutions Private Limited       Bangalore   
2                         Hot Stuffs Private Limited          Mumbai   
3    NxtWave Disruptive Technologies Private Limited  Work from home   
4             ReThought Technologies Private Limited           Delhi   
..                                               ...             ...   
404                Mohite Consultancy Services (MCS)     Koramangala   
405                                        OpeninApp       Bangalore   
406                                       Confluencr       Bangalore   
407                                    Nerdy Academy          Mumbai   
408                   Fabulous Media Private Limited         Gurgaon   

             Start Date           CTC (Annual) Experience   Posted Ago Role  
0    Starts Immediately  ₹ 2,00,000 - 2,40,000  0-2 years  3 weeks ago  Job  
1    Starts Immediately  ₹ 3,00,000 - 5,00,000  0-5 years    1 day ago  Job  
2    Starts Immediately  ₹ 2,10,000 - 3,00,000  0-2 years   6 days ago  Job  
3    Starts Immediately  ₹ 3,00,000 - 6,00,000  0-5 years  2 weeks ago  Job  
4    Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years  2 weeks ago  Job  
..                  ...                    ...        ...          ...  ...  
404  Starts Immediately  ₹ 2,16,000 - 2,40,000  0-2 years   1 week ago  Job  
405  Starts Immediately  ₹ 2,00,000 - 8,00,000  0-5 years   1 week ago  Job  
406  Starts Immediately  ₹ 3,00,000 - 4,50,000  0-2 years   6 days ago  Job  
407  Starts Immediately  ₹ 2,40,000 - 5,80,000  0-5 years  3 weeks ago  Job  
408  Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years   5 days ago  Job  

[409 rows x 9 columns]

Unnamed: 0                                        Position  \
0             0                        Junior Interior Designer   
1             1  Business Development Representative - B2B SaaS   
2             2                      Corporate Sales Specialist   
3             3                  Career Guidance Expert - Sales   
4             4                           Video Content Creator   
..          ...                                             ...   
404         404                  Business Development Executive   
405         405                           Engagement Strategist   
406         406                    Brand Partnerships Associate   
407         407                  Business Development Executive   
408         408                Junior Lead Generation Associate   

                                        Company Name        Location  \
0                           Dexterio Private Limited          Jaipur   
1            Zylu Business Solutions Private Limited       Bangalore   
2                         Hot Stuffs Private Limited          Mumbai   
3    NxtWave Disruptive Technologies Private Limited  Work from home   
4             ReThought Technologies Private Limited           Delhi   
..                                               ...             ...   
404                Mohite Consultancy Services (MCS)     Koramangala   
405                                        OpeninApp       Bangalore   
406                                       Confluencr       Bangalore   
407                                    Nerdy Academy          Mumbai   
408                   Fabulous Media Private Limited         Gurgaon   

             Start Date           CTC (Annual) Experience   Posted Ago Role  
0    Starts Immediately  ₹ 2,00,000 - 2,40,000  0-2 years  3 weeks ago  Job  
1    Starts Immediately  ₹ 3,00,000 - 5,00,000  0-5 years    1 day ago  Job  
2    Starts Immediately  ₹ 2,10,000 - 3,00,000  0-2 years   6 days ago  Job  
3    Starts Immediately  ₹ 3,00,000 - 6,00,000  0-5 years  2 weeks ago  Job  
4    Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years  2 weeks ago  Job  
..                  ...                    ...        ...          ...  ...  
404  Starts Immediately  ₹ 2,16,000 - 2,40,000  0-2 years   1 week ago  Job  
405  Starts Immediately  ₹ 2,00,000 - 8,00,000  0-5 years   1 week ago  Job  
406  Starts Immediately  ₹ 3,00,000 - 4,50,000  0-2 years   6 days ago  Job  
407  Starts Immediately  ₹ 2,40,000 - 5,80,000  0-5 years  3 weeks ago  Job  
408  Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years   5 days ago  Job  

[409 rows x 9 columns]

Unnamed: 0                                        Position  \
0             0                        Junior Interior Designer   
1             1  Business Development Representative - B2B SaaS   
2             2                      Corporate Sales Specialist   
3             3                  Career Guidance Expert - Sales   
4             4                           Video Content Creator   
..          ...                                             ...   
404         404                  Business Development Executive   
405         405                           Engagement Strategist   
406         406                    Brand Partnerships Associate   
407         407                  Business Development Executive   
408         408                Junior Lead Generation Associate   

                                        Company Name        Location  \
0                           Dexterio Private Limited          Jaipur   
1            Zylu Business Solutions Private Limited       Bangalore   
2                         Hot Stuffs Private Limited          Mumbai   
3    NxtWave Disruptive Technologies Private Limited  Work from home   
4             ReThought Technologies Private Limited           Delhi   
..                                               ...             ...   
404                Mohite Consultancy Services (MCS)     Koramangala   
405                                        OpeninApp       Bangalore   
406                                       Confluencr       Bangalore   
407                                    Nerdy Academy          Mumbai   
408                   Fabulous Media Private Limited         Gurgaon   

             Start Date           CTC (Annual) Experience   Posted Ago Role  
0    Starts Immediately  ₹ 2,00,000 - 2,40,000  0-2 years  3 weeks ago  Job  
1    Starts Immediately  ₹ 3,00,000 - 5,00,000  0-5 years    1 day ago  Job  
2    Starts Immediately  ₹ 2,10,000 - 3,00,000  0-2 years   6 days ago  Job  
3    Starts Immediately  ₹ 3,00,000 - 6,00,000  0-5 years  2 weeks ago  Job  
4    Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years  2 weeks ago  Job  
..                  ...                    ...        ...          ...  ...  
404  Starts Immediately  ₹ 2,16,000 - 2,40,000  0-2 years   1 week ago  Job  
405  Starts Immediately  ₹ 2,00,000 - 8,00,000  0-5 years   1 week ago  Job  
406  Starts Immediately  ₹ 3,00,000 - 4,50,000  0-2 years   6 days ago  Job  
407  Starts Immediately  ₹ 2,40,000 - 5,80,000  0-5 years  3 weeks ago  Job  
408  Starts Immediately  ₹ 2,00,000 - 3,00,000  0-2 years   5 days ago  Job  

[409 rows x 9 columns]

# Missing Values

In [11]:
df.isnull().sum()

Unnamed: 0      0
Position        0
Company Name    0
Location        0
Start Date      0
CTC (Annual)    0
Experience      0
Posted Ago      0
Role            0
dtype: int64

# Invalid Values

# Outliers

In [12]:
import matplotlib.pyplot as plt
%matplotlib inline

import seaborn as sns